javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesScenario.java

```diff
@@ -0,0 +1,578 @@
+     */
+    public static UserNames createVpcUsers() {
+        /*
+        Optionally create a VPC.
+        Create two IAM users, one with S3 Express One Zone permissions and one without.
+        */
+        logger.info("""
+            1. First, we'll set up a new VPC and VPC Endpoint if this program is running in an EC2 instance in the same AZ as your\s
+            Directory buckets will be. Are you running this in an EC2 instance located in the same AZ as your intended Directory buckets?
+            """);
+
+        logger.info("Do you want to setup a VPC Endpoint? (y/n)");
+        String endpointAns = scanner.nextLine().trim();
+        if (endpointAns.equalsIgnoreCase("y")) {
+            logger.info("""
+                Great! Let's set up a VPC, retrieve the Route Table from it, and create a VPC Endpoint to connect the S3 Client to.
+                """);
+            try {
+                s3DirectoriesActions.setupVPCAsync().join();
+            } catch (CompletionException ce) {
+                Throwable cause = ce.getCause();
+                if (cause instanceof Ec2Exception) {
+                    logger.error("IamException occurred: {}", cause.getMessage(), ce);
+                } else {
+                    logger.error("An unexpected error occurred: {}", cause.getMessage(), ce);
+                }
+            }
+            waitForInputToContinue(scanner);
+        } else {
+            logger.info("Skipping the VPC setup. Don't forget to use this in production!");
+        }
+        logger.info("""            
+            2. Policies, users, and roles with CDK.
+            Now, we'll set up some policies, roles, and a user. This user will 
+            only have permissions to do S3 Express One Zone actions.
+            """);
+
+        waitForInputToContinue(scanner);
+        logger.info("Use AWS CloudFormation to create IAM roles that is required for this scenario.");
+        CloudFormationHelper.deployCloudFormationStack(stackName);
+        Map<String, String> stackOutputs = CloudFormationHelper.getStackOutputsAsync(stackName).join();
+
+        // Create two users required for this scenario.
+        regularUser = stackOutputs.get("RegularUser");
+        expressUser = stackOutputs.get("ExpressUser");
+
+        UserNames names = new UserNames();
+        names.setRegularUserName(regularUser);
+        names.setExpressUserName(expressUser);
+        return names;
+    }
+
+    /**
+     * Sets up the necessary clients and buckets for the S3 Express service.
+     *
+     * @param expressUserName the username for the user with S3 Express permissions
+     * @param regularUserName the username for the user with regular S3 permissions
+     */
+    public static void setupClientsAndBuckets(String expressUserName, String regularUserName) {
+        Scanner locscanner = new Scanner(System.in);
+        String accessKeyIdforRegUser;
+        String secretAccessforRegUser;
+        try {
```

Comment:
There should be a separate method to create the access keys and return the Access key, so it can be reused and avoid the duplication.

Reply:
Following the spec and Python app for this. The code is in its own method in the S3DirectoriesActions class.